src/chat/chatReducers.js

```diff
@@ -43,6 +43,19 @@ export default (state: ChatState = initialState, action: Action) => {
       };
 
     case SWITCH_NARROW: {
+      if (action.message) {
+        const key = JSON.stringify(action.narrow);
+        return {
+          ...state,
+          narrow: action.narrow,
+          fetchingOlder: false,
+          fetchingNewer: false,
+          messages: {
+            ...state.messages,
+            [key]: [action.message],
```

Comment:
What happens if the message already exists?
Maybe not mutate messages state at all?
Is there a case where the message will not be there already?